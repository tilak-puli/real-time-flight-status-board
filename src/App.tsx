import "./App.css";
import { Routes, Route } from "react-router-dom";

// Fonts for Material UI components
import "@fontsource/roboto/300.css";
import "@fontsource/roboto/400.css";
import "@fontsource/roboto/500.css";
import "@fontsource/roboto/700.css";
import theme from "./style/theme";
import { ThemeProvider } from "@mui/material/styles";
import WithHeader from "./layout/WithHeader";
import FlightsList from "./screens/FlightsList/FlightsList";
import FlightDetails from "./screens/FlightDetails/FlightDetails";

function App() {
  return (
    <ThemeProvider theme={theme}>
      <Routes>
        <Route path="/" element={WithHeader(<FlightsList />)} />
        <Route
          path="/flight-details/:id"
          element={WithHeader(<FlightDetails />)}
        />
      </Routes>
    </ThemeProvider>
  );
}

export default App;
