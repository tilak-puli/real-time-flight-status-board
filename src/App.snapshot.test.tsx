import renderer from "react-test-renderer";

import App from "./App";
import { BrowserRouter } from "react-router-dom";
import MockAdapter from "axios-mock-adapter";
import instance from "./API/axios";
const mock = new MockAdapter(instance);

describe("App", () => {
  it("renders flight list by default", () => {
    const dummyFlights = [
      {
        flightNumber: "123",
        id: 1,
        airline: "airline1",
        origin: "origin1",
        destination: "destination",
        departureTime: "2023-09-11T18:00:51.613Z",
        status: "departed",
      },
    ];
    mock.onGet("/flights").reply(200, dummyFlights);

    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
