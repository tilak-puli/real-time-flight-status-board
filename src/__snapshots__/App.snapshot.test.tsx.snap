// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders flight list by default 1`] = `
<div
  className="css-zf0iqh"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1gedkir-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ap1hwn-MuiToolbar-root"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="MuiTypography-root MuiTypography-h6 css-817iol-MuiTypography-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-6z2nyb-MuiSvgIcon-root"
            data-testid="AirlinesIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 4 2 20h17l3-16h-9zm1.5 10c-1.38 0-2.5-1.12-2.5-2.5S13.12 9 14.5 9s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          Flight Status Board
        </div>
      </a>
    </div>
  </header>
  <div
    className="css-1r3ldka"
  >
    <div />
  </div>
</div>
`;
